eng/Versions.props

```diff
@@ -8,4 +8,16 @@
     <PreReleaseVersionLabel>alpha</PreReleaseVersionLabel>
     <PreReleaseVersionIteration>1</PreReleaseVersionIteration>
   </PropertyGroup>
+
+  <PropertyGroup>
+    <AzureActiveDirectoryIdentityModelExtensionsVersion>7.1.2</AzureActiveDirectoryIdentityModelExtensionsVersion>
```

Comment:
The original issue that prompted the logging of https://github.com/dotnet/source-build/issues/3897 was https://github.com/dotnet/source-build-externals/pull/248/files/e22d3d81f27c6b8389d97c80819b86b296ed1df4.  When I made that comment, the intent was to encapsulate the reference package versions in a single location to help ensure the repo's only referenced a single version and to make maintenance easier. This resulting issue pivoted and suggested encapsulating the external package versions as you did. 

Using a similar/same naming convention as the reference packages introduces a potential issue.  If the names match the generated artifact package version property names then anything defined here in the Versions.props file will get overridden during a product source-build with the N-1 version.  This can therefore cause source-build to product an older external package version than intended.

Additionally names like DockerCredsProvider220PackageVersion are actually adding more maintenance in the upgrade scenario because not only does the value have to change, the property name does as well.

I see a few options:

1. Abandon these changes
2. Change the package naming convention to ensure a collisions with the N-1 packages never occur.  e.g. introduce a new suffix - ExternalPackageVersion.  We accept that  some property names will need to change with each package upgrade.

With either approach, the versions of the package references owned by this repo should be encapsulated in the versions.props file e.g. https://github.com/dotnet/source-build-externals/blob/main/eng/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/Microsoft.DotNet.SourceBuild.Tasks.XPlat.csproj#L10

Reply:
Thank you for that context.

Which option would you prefer?

I tried implementing option 2, except I left the docker-creds-provider version hardcoded outside of eng/Versions.props.

Reply:
I personally have never experienced a need for these.  From a conversation with @mthalman, he sees value in them.  I have no objection to them as long as the aforementioned n-1 version issue is addressed.